polarCoordinateTraversal_dotprod.m

```diff
@@ -0,0 +1,198 @@
+% Input:
+%    min_bound: The lower left corner of the bounding box.
+%    max_bound: The upper right corner of the bounding box.
+%    ray origin: The origin of the ray in (x, y) coordinates.
+%    ray direction: The direction of the ray in (x, y) coordinates.
+%    circle_center: The x, y location of the center of the circle.
+%    circle_max_radius: The largest that encompasses the circle.
+%    num_radial_sections: The number of radial sections in the circle.
+%    num_angular_sections: The number of angular sections in the circle.
+%    t_begin: The beginning time of the ray.
+%    t_end: The end time of the ray.
+%
+% Requires:
+%    max_bound > min_bound
+%    circle center is within max_bound and min_bound.
+%    t_end > t_begin >= 0.0
+%    circle_max_radius > 0
+%    num_radial_sections > 0
+%    num_angular_sections > 0
+%
+% Returns:
+%    radial_voxels: A list of the radial voxels that were hit by the ray.
+%    angular_voxels: A list of the angular voxels that were hit by the ray.
+%    These lists, used in conjunction, will produce the path of the ray
+%    through the voxels using each point. For example,
+%    [radial_voxels(1), angular_voxels(1)] is the first voxel the ray
+%    travels through.
+%
+% Notes:
+%    Currently under construction.
+close all;
+circle_center_x = circle_center(1);
+circle_center_y = circle_center(2);
+ray_origin_x = ray_origin(1);
+ray_origin_y = ray_origin(2);
+ray_direction_x = ray_direction(1);
+ray_direction_y = ray_direction(2);
+
+min_bound_x = min_bound(1);
+min_bound_y = min_bound(2);
+max_bound_x = max_bound(1);
+max_bound_y = max_bound(2);
+
+ray_start = ray_origin + t_begin * ray_direction;
+ray_start_x = ray_start(1);
+ray_start_y = ray_start(2);
+
+ray_end = ray_origin + t_end * ray_direction;
+ray_end_x = ray_end(1);
+ray_end_y = ray_end(2);
+
+if (verbose)
+    figure;
+    hold on;
+    title('Polar Coordinate Voxel Traversal')
+    
+    if (t_begin ~= 0.0)
+        % Mark the ray origin if the time does not start at 0.0
+        text(ray_origin_x, ray_origin_y, ' ray origin');
+        plot(ray_origin_x, ray_origin_y, 'k.', 'MarkerSize', 10);
+        quiver(ray_origin_x, ray_origin_y, ray_direction_x, ray_direction_y, t_begin - 0.0, 'LineWidth', 1.5);
+    end
+    
+    % Draw the ray.
+    text(ray_start_x, ray_start_y, ' ray start');
+    text(ray_end_x, ray_end_y, ' ray end');
+    plot(ray_end_x, ray_end_y, 'k.', 'MarkerSize', 10);
+    plot(ray_start_x, ray_start_y, 'k.', 'MarkerSize', 10);
+    quiver(ray_start_x, ray_start_y, ray_direction_x, ray_direction_y, t_end - t_begin, 'LineWidth', 1.5);
+    
+    % Draw the axis.
+    axis tight;
+    xlim([min_bound_x, max_bound_x]);
+    ylim([min_bound_y, max_bound_y]);
+    xlabel('x');
+    ylabel('y');
+    grid on;
+    
+    % Draw the radial sections.
+    current_max_radius = circle_max_radius;
+    delta_radius = circle_max_radius / num_radial_sections;
```

Comment:
Since this is defined in an ```if verbose ... end``` block, if verbose is false, ```delta_radius``` will not be defined. Move this line outside of the if block so that we can use ```delta_radius``` on line 107..

Reply:
> Edit: Adding a few small changes from last night, I get the following for the radial traversal:

To be clear, do both traversals (orig. & dot) give this result?

Reply:
No. original gives the correct result (at least when the ray goes through the origin as in this example). The dotprod version gives an incorrect result.

Reply:
OK. Because we are hoping to get this to Matt tomorrow, what should we do? Address this problem in dot version or go back to original?

Reply:
Let's stick with this for now. Visually it seems fine, just need to work out some kinks.

Reply:
Bueno